include/ringmesh/basic/geometry.h

```diff
@@ -290,6 +306,22 @@ namespace RINGMesh
         explicit PlaneReferenceFrame3D( const Geometry::Plane& plane );
     };
 
+    /*!
+     * @brief Reference frame aligned along the plane normal and whose u axis is
+     * upward
+     */
+    struct RINGMESH_API LineReferenceFrame2D : public ReferenceFrame< 2 >
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Promote this "struct" to a "class" or remove its member functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1771)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this redundant access specifier; it is already "public". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3540)